src/main/java/com/google/sps/data/DBInterface.java

```diff
@@ -146,6 +169,49 @@
    */
   public void deleteUserProperty(String userId, String objectId, String collection);
 
+  /**
+   * Deletes a property value for a certain user document. Can be used to let a user delete a recipe
+   * from saved or created, or to let user unfollow a tag.
+   * Done within a transaction.
+   *
+   * @param userId the user's Firebase ID
+   * @param objectId the ID of either a recipe if the intent is to unsave/create, or of a tag for
+   *     tag unfollowing.
+   * @param collection a KEY constant from User class indicating which mode -- save, create, or tag.
+   */
+  public void deleteUserProperty(String userId, String objectId, String collection, Transaction t);
+
+  /**
+   * Checks if a specified user has a given recipe as true in a given map field.
+   *
+   * @param userId the user's Firebase ID.
+   * @param recipeId the recipe's Firebase ID.
+   * @return true if it is in the map as true, false if in map as false, null if not in map or user
+   *     does not exist
+   */
+  public Boolean inUserMap(String userId, String recipeId, String mapName);
+
+  /**
+   * inUserMap, but for arrays
+   *
+   * @param userId the user's Firebase ID.
+   * @param recipeIds the recipes Firebase IDs.
+   * @return array of [true if it is in the map as true, false if in map as false, null if not in
+   *     map or user does not exist]
+   */
+  public Boolean[] inUserMap(String userId, String[] recipeIds, String mapName);
```

Comment:
I wonder if it is better to return an String[] with users who are not in the map filtered out.

Reply:
The main use of this is to get whether a list of posts has been voted on / saved in the feed page. While your idea would work for the saved posts, the voted posts have 3 options (upvoted, downvoted, and not voted). I suppose making it return a map would also work, but that makes the function a little more complex.

Reply:
Maybe renaming this function to `getUserPropertyInMap` would be a better idea.

Reply:
renamed to `getUserProperty`